src/hotspot/share/classfile/dictionary.cpp

```diff
@@ -424,16 +401,15 @@ void Dictionary::validate_protection_domain(unsigned int name_hash,
   // and protection domain are expected to succeed.
   {
     MutexLocker mu(THREAD, SystemDictionary_lock);
-    int d_index = hash_to_index(name_hash);
-    add_protection_domain(d_index, name_hash, klass,
+    add_protection_domain(THREAD, klass,
                           protection_domain);
   }
 }
 
 // During class loading we may have cached a protection domain that has
 // since been unreferenced, so this entry should be cleared.
 void Dictionary::clean_cached_protection_domains(GrowableArray<ProtectionDomainEntry*>* delete_list) {
-  assert(Thread::current()->is_Java_thread(), "only called by JavaThread");
+  assert(JavaThread::current()->is_Java_thread(), "only called by JavaThread");
```

Comment:
`JavaThread::current` already does this check, so just leave it as `Thread::current()`.